dfir_rs/src/scheduled/graph.rs

```diff
@@ -282,8 +291,26 @@ impl<'a> Dfir<'a> {
         'pop: while let Some(sg_id) =
             self.context.stratum_queues[self.context.current_stratum].pop_front()
         {
+            let sg_data = &mut self.subgraphs[sg_id];
+
+            // Update handoff metadata.
+            // NOTE(mingwei):
+            // This is done BEFORE running the subgraph because we rely on the fact that subgraphs always(*) consume
+            // all inputs. This is NOT done AFTER running the subgraph because it may send to the handoff multiple
```

Comment:
Comment could be clearer. I *think* you're saying (a) we can get reliable input counts from a BEFORE check, and (b) an AFTER check is unreliable, but the consumer will make it right when they do their BEFORE check?

I.e.  we measure at input, not at output?